### PR TITLE
Use default bash login PATH

### LINKED ISSUE #2065: Headers not found on iOS catalina
Copy pasting the errors I've seen so far so it's easy to search on the repo/outside this repo:
```
'errno.h' file not found
...
Failed to run custom build command for `libloading v0.5.2`
        'pthread.h' file not found
```

I've filed an issue upstream at https://github.com/alexcrichton/cc-rs/issues/452 where you can find more details.

┆Issue is synchronized with this [Jira Story](https://jira.allizom.org/browse/SYNC-628)
┆Story Points: 3

### PR DESCRIPTION
**Pull request by eoger**
Fixes #2065.

aaaah

### CODE REVIEW

File: build-scripts/xc-universal-binary.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
 set -euvx
 
+# XCode tries to be helpful and overwrites the PATH. Reset that.
+PATH="$(bash -l -c 'echo $PATH')"
```
Comment (thomcc) [inline build-scripts/xc-universal-binary.sh]: Have we tested that this doesn't break anything locally?

Comment (eoger) [reply build-scripts/xc-universal-binary.sh]: WFM locally.

Comment (thomcc) [reply build-scripts/xc-universal-binary.sh]: This worries me that we're going to be using a different clang than the one bundled with xcode that swift links against (swift links against libclang for various reasons like parsing C++ headers and such), but maybe it's fine...

Comment (eoger) [reply build-scripts/xc-universal-binary.sh]: Agree, I'm not a huge fan of the solution either but I couldn't find anything wrong with it so far.

**Pull request merged**